### PR TITLE
Log exceptions atomically

### PR DESCRIPTION
**Pull request by sj26**
When distributed over multiple logger calls the lines can become intermixed with other log statements. Combining them into a single logger call makes sure they always get logged together.

### CODE REVIEW

File: actionpack/lib/action_dispatch/middleware/debug_exceptions.rb [hunk 1]
```diff
@@ -180,11 +180,14 @@ def log_error(request, wrapper)
...
         ActiveSupport::Deprecation.silence do
-          logger.fatal "  "
-          logger.fatal "#{exception.class} (#{exception.message}):"
-          log_array logger, exception.annoted_source_code if exception.respond_to?(:annoted_source_code)
-          logger.fatal "  "
-          log_array logger, trace
+          message = []
+          message << "  "
+          message << "#{exception.class} (#{exception.message}):"
+          message += exception.annoted_source_code if exception.respond_to?(:annoted_source_code)
```
[the remark this answers is not in the document]
Comment (sj26) [reply actionpack/lib/action_dispatch/middleware/debug_exceptions.rb]: I believe it is an array? The previous version implied it was.

[the remark this answers is not in the document]
Comment (rafaelfranca) [reply actionpack/lib/action_dispatch/middleware/debug_exceptions.rb]: Duh! Yeah, that makes sense, can we use `concat` instead? Don't think we need to allocate other array.

[the remark this answers is not in the document]
Comment (sj26) [reply actionpack/lib/action_dispatch/middleware/debug_exceptions.rb]: Absolutely, done! I agonised a little over how to construct an array, then decided to just do something which aligned prettily and just wait for feedback. 😅

[the remark this answers is not in the document]
Comment (sj26) [reply actionpack/lib/action_dispatch/middleware/debug_exceptions.rb]: I think we can avoid all those concats, too, I just don't know how I feel about the syntax:

```ruby
ActiveSupport::Deprecation.silence do
  log_array(logger, [
    "  ",
    "#{exception.class} (#{exception.message}):",
    *(exception.annoted_source_code if exception.respond_to?(:annoted_source_code)),
    "  ",
    *trace
  ])
end
```

**Pull request merged**